agent/config/types.go

```diff
@@ -110,7 +110,7 @@ type Config struct {
 	TaskIAMRoleEnabled bool
 
 	// TaskCPUMemLimit specifies if Agent can launch a task with a hierarchical cgroup
-	TaskCPUMemLimit bool
+	TaskCPUMemLimit DefaultBool
```

Comment:
Why use a new type and not just `*bool`?

Reply:
Using a `*bool` to effectively make a boolean have 3 states made me feel sad. I don't have a good technical reason. Is there a precedent?